### PR TITLE
Feature/regionProgram

### PR DESCRIPTION
**Pull request by Azher2Ali**
In this PR, we are replacing the `Region` as this is being replaced by the Data Center entity and relationship and therefore we have removed Region from the Program entity and the Region tables from the DB and also resolved the issues related to previous ticket #412 by modify the test cases to include a dataCenterId

Attaching Rest test response and gRPC test response for reference.

Rest Response of list Programs :
Screenshot 2023-09-29 at 3 07 10 PM

gRPC Response of list Programs :
Screenshot 2023-09-29 at 3 09 00 PM

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/model/dto/builder/ProgramDTOBuilder.java [hunk 1]
```diff
@@ -1,35 +1,33 @@
...
+    programDTO.setProgramCancers(
+        canerDtoBuilder.convertEntityToDTO(programEntity.getProgramCancers()));
+    programDTO.setProgramCountries(
+        countryDTOBuilder.convertEntityToDTO(programEntity.getProgramCountries()));
+    programDTO.setProgramPrimarySites(
+        primarySiteDTOBuilder.convertEntityToDTO(programEntity.getProgramPrimarySites()));
+    programDTO.setProgramInstitutions(
+        institutionDTOBuilder.convertEntityToDTO(programEntity.getProgramInstitutions()));
+    return programDTO;
+  }
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/model/dto/builder/ProgramDTOBuilder.java:32]: You might consider making the programDTO a Builder using lombok. Definitely not required, nor a big difference, but might be cleaner and not require instantiating partial objects:
https://projectlombok.org/features/Builder

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java [hunk 2]
```diff
@@ -1,488 +1,460 @@
...
+  @Autowired
+  public ProgramService(
+      @NonNull ProgramRepository programRepository,
+      @NonNull CancerRepository cancerRepository,
+      @NonNull PrimarySiteRepository primarySiteRepository,
+      @NonNull ProgramConverter programConverter,
+      @NonNull ProgramCancerRepository programCancerRepository,
+      @NonNull ProgramPrimarySiteRepository programPrimarySiteRepository,
+      @NonNull InstitutionRepository institutionRepository,
+      @NonNull RegionRepository regionRepository,
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/services/ProgramService.java:88]: as we are removing region, is this still needed?

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java [hunk 3]
```diff
@@ -1,488 +1,460 @@
...
+      @NonNull List<String> institutions,
+      @NonNull List<String> countries)
+      throws NotFoundException {
+    if (cancers.isEmpty()
+        || primarySites.isEmpty()
+        || institutions.isEmpty()
+        || countries.isEmpty()) {
+      throw Status.INVALID_ARGUMENT
+          .augmentDescription(
+              "Cannot update program. Cancer, primary site, institution, country, and region cannot be empty.")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/services/ProgramService.java:223]: Shouldnt need to enforce region now, let's update the message.

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java [hunk 4]
```diff
@@ -1,488 +1,460 @@
...
+    return List.copyOf(primarySiteRepository.findAll());
+  }
+
+  List<CountryEntity> listCountries() {
+    return List.copyOf(countryRepository.findAll());
+  }
+
+  List<RegionEntity> listRegions() {
+    return List.copyOf(regionRepository.findAll());
+  }
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/services/ProgramService.java:415]: Still needed?

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java (hunk 2 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/services/ProgramService.java:87]: These are still being used through grpc endpoints for the getRegions RPC

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java (hunk 3 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/services/ProgramService.java]: updated the message

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java (hunk 4 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/services/ProgramService.java:412]: listRegions() method is being used by getRegions RPC to fetch the regions

File: src/main/java/org/icgc/argo/program_service/model/dto/builder/ProgramDTOBuilder.java (hunk 1 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/model/dto/builder/ProgramDTOBuilder.java:32]: These are existing methods and have modified only the removal of region mapping against the Entity and Pojo.

**Review by joneubank — APPROVED** [review]

**Pull request merged**